client/mysql_plugin.c

```diff
@@ -1168,7 +1170,9 @@ static int dump_bootstrap_file(char *bootstrap_file)
 
 static int bootstrap_server(char *server_path, char *bootstrap_file)
 {
-  char bootstrap_cmd[FN_REFLEN];
+  const char format_cmd[]= "%s --no-defaults --bootstrap --datadir=%s "
+                           "--basedir=%s < %s";
+  char bootstrap_cmd[FN_REFLEN * 2 + sizeof(format_cmd)];
```

Comment:
`FN_REFLEN * 4` ?

Reply:
`opt_datadir` and `opt_basedir` have no size of FN_REFLEN, they are dynamic, so `strlen(opt_datadir)` + `strlen(opt_basedir)`